system/lib/libc/musl/src/thread/pthread_barrier_wait.c

```diff
@@ -87,14 +87,29 @@ int pthread_barrier_wait(pthread_barrier_t *b)
 		while (spins-- && !inst->finished)
 			a_spin();
 		a_inc(&inst->finished);
-		while (inst->finished == 1) {
 #ifdef __EMSCRIPTEN__
-			emscripten_futex_wait(&inst->finished, 1, INFINITY);
+		int is_main_thread = emscripten_is_main_runtime_thread();
```

Comment:
Should this not be `emscripten_is_main_browser_thread`.    IIUC its only the browser thread that can't block.   This would also seem to match __timedwait.c.

Reply:
I remembered a comment about `emscripten_is_main_browser_thread()` versus `emscripten_is_main_runtime_thread()`, see: https://github.com/emscripten-core/emscripten/pull/12963#discussion_r553854443.

Therefore shouldn't the `__timedwait_cp` function instead check for `emscripten_is_main_runtime_thread()` since that function is a cancellation point?

(note that `emscripten_is_main_browser_thread()` also returns `true` on non-worker threads on Node, I attempted [to change this](https://github.com/kleisauke/emscripten/commit/36236cf4aeccdca53cb5e4aff5ea014a1d569250) but unfortunately it didn't work out)

Reply:
Well it seems that both `pthread_barrier_wait.c` and `__timedwait.c` should use the same thing.   Having them differ seems bad.  Perhaps land this change with `emscripten_is_main_browser_thread` and then we can consider a separate PR to change them both to `emscripten_is_main_runtime_thread` as a followup?

Reply:
Good, catch. Fixed.